hivelocity/instances_v2.go

```diff
@@ -82,60 +85,93 @@ func getHivelocityDeviceIDFromNode(node *corev1.Node) (int32, error) {
 	return int32(deviceID), nil
 }
 
+// lookUpDevice looks for device via Hivelocity API if provider ID is present otherwise look for machine name label
+// present in the devices (caphv-machine-name=foo).
+func (i2 *HVInstancesV2) lookUpDevice(ctx context.Context, node *corev1.Node) (*hv.BareMetalDevice, error) {
+	if node.Spec.ProviderID != "" {
+		deviceID, err := getHivelocityDeviceIDFromNode(node)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"[InstanceExists] getHivelocityDeviceIDFromNode() failed. node %q: %w",
+				node.GetName(),
+				err,
+			)
+		}
+		device, err := i2.client.GetBareMetalDevice(ctx, deviceID)
+		if errors.Is(err, client.ErrNoSuchDevice) {
+			return nil, nil
+		}
+		if err != nil {
+			return nil, fmt.Errorf(
+				"[InstanceExists] GetBareMetalDevice() failed. node %q, deviceID %d: %w",
+				node.GetName(),
+				deviceID,
+				err,
+			)
+		}
+		return device, nil
+	} else {
+		devices, err := i2.client.ListDevice(ctx)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"[InstanceExists] ListDevice() failed. node %q: %w",
+				node.GetName(),
+				err,
+			)
+		}
+
+		for _, device := range devices {
+			name, err := hvutils.GetMachineNameFromTags(device.Tags)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"InstanceExists(): GetMachineNameFromTags() failed. node %q: %w",
+					node.GetName(),
+					err,
+				)
+			}
+
+			if name == node.GetName() {
+				return &device, nil
+			}
+		}
+
+		return nil, client.ErrNoSuchDevice
```

Comment:
here you return an error, and on top you return nil, nil.. From what I see how you use it in InstanceExists I don't think you should return an error here...